proposals/vector_to_regular_points.json

```diff
@@ -0,0 +1,41 @@
+{
+    "id": "vector_to_regular_points",
+    "summary": "Sample regular points from geometries",
+    "description": "Generate a vector data cube of points by sampling regularly-spaced points from input geometries.",
+    "categories": [
+        "cubes",
+        "vector"
+    ],
+    "experimental": true,
+    "parameters": [
+        {
+            "name": "data",
+            "description": "Input geometries for sample extraction. Vector properties are preserved for vector data cubes and all GeoJSON Features.\n\nTo maximize interoperability, a nested `GeometryCollection` should be avoided. Furthermore, a `GeometryCollection` composed of a single type of geometries should be avoided in favour of the corresponding multi-part type (e.g. `MultiPolygon`).",
+            "schema": [
+                {
+                    "type": "object",
+                    "subtype": "geojson"
+                },
+                {
+                    "type": "object",
+                    "subtype": "vector-cube"
+                }
+            ]
+        },
+        {
+            "name": "cell_size",
+            "description": "The size in degrees of the cell of the regular grid. If the cell size is bigger than the geometry, a single point is returned located in the centroid of the geometry. If the centroid is not enclosed in the geometry area, no point is returned.",
```

Comment:
I've just checked and the answer is yes, this is fine for example:
```
{
  "type": "FeatureCollection",
  "features": [
    {
      "type": "Feature",
      "properties": {},
      "geometry": null
    }]
}
```

source: https://datatracker.ietf.org/doc/html/rfc7946 (look for the Feature keyword)

Reply:
Should we add that we would need to return a Feature with empty geometry in that particular case?

Reply:
```suggestion
            "description": "The size in degrees of the cell of the regular grid. If the cell size is bigger than the geometry, a single point is returned located in the centroid of the geometry. If the centroid is not enclosed in the geometry area, no point is returned (i.e. a feature without geometry).",
```
Like that?

Another question: How does this description work for Point and LineString Geometries as `data`? Or does this process only really work for  (Multi)Polygons?

Reply:
Good question. I would return points as they are (same as in the random sampling), since if someone included them in the input data could mean that they want to have them in output as well. Or should we have a boolean for that? Like "return single Point even if not regularly spaced?".
For lines I think it would be possible to sample points with a spacing depending on the cell. If the spacing is greater than the line length we can again either return the center or nothing.

There are so many combinations to consider actually...

Reply:
Discussed with @clausmichele: We propose the following per geometry:
1. For Polygons: As described above.
2. For Lines: Start with a point at the beginning on the line and then put a new point every `cell_size` degrees (by the way, should we rename cell_size to distance or so?)
3. For points simply return the point.

Reply:
add smth like: distance: minimum distance to other points per geometry